### ISSUE TITLE
model conversion, conversion issues on different deep learning framework

### ISSUE THREAD

**Issue by kyunghwan** [openvinotoolkit/openvino#7760]
- OpenVINO => 2021.4
- Operating System / Platform => Ubuntu 20.04 ,Myriad NCS2
- Problem classification: Model Conversion
- Framework: Mxnet 1.8.0, Pytorch 1.8.0, onnx 1.10.1
- Model name: Arcface(resnet100)

Hi.

I am using arcface architecture.
Pre-trained model(mxnet)   -----mo_mxnet.py------> work well
Pre-trained model(onnx)     -----torch2onnx-----mo_onnx.py------> model transfer is done, but strange result
Pre-trained model(Pytorch) -----torch2onnx-----mo_onnx.py------> model transfer is done, but strange result

The mxnet model performs normalization within the model, but the torch model does not.
So, normalization was performed before running the torch model.
I don't know why the problem occurs. 
Are there any factors to consider in this process?

**Comment by Iffa-Meah** [comment]
Hi,

is it possible for you to share your model files?
Any chance you had tried the model with **mo.py** instead to see if there are any differences? This is a general model optimizer that should work with all models.

**Comment by kyunghwan** [comment]
@Iffa-Meah 

Thank you for review

https://drive.google.com/drive/folders/1kiuL-1et9YNDGQEK9W9umPvpSzF7wjZW?usp=sharing

I downloaded the pre-trained model (https://github.com/deepinsight/insightface)
I used mo.py, and set input size [1,3,112,112] option

**Comment by Iffa-Meah** [comment]
It would be great if you could share your model files for us to investigate further.
Could you share the full command that you had used to convert the model? 

The --scale flag required to be passed to the model optimizer especially for IR to be used with MYRIAD/NCS2.
Plus, maybe the --reverse_input_channels is needed. Check [here](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model_General.html#when_to_reverse_input_channels) to see whether you need to use it.

**Comment by kyunghwan** [comment]
@Iffa-Meah
Thank you for review

https://drive.google.com/drive/folders/1kiuL-1et9YNDGQEK9W9umPvpSzF7wjZW?usp=sharing

I think setting the scale option is the right way.
Pretrained model is ok if I set scale option
But, custom trained model is a difference between the torch model and the openvino model.
I tried the reverse channel option, but it didn't work.
Is the problem caused by the FP32 torch model ?

arcface_r100_pretrained_frommxnet.zip
arcface_r100_pretrained_fromtorch.zip
python3 mo.py --input_model model.onnx --output_dir ./ --input_shape [1,3,112,112] --data_type FP16 

arcface_r100_pretrained_fromtorch_scaleoption.zip
arcface_r100_customtrained_fromtorch_scaleoption.zip
python3 mo.py --input_model model.onnx --output_dir ./ --input_shape [1,3,112,112] --data_type FP16 --mean_values [127.5] --scale_values [128]

**Comment by Iffa-Meah** [comment]
The data format of the native model shouldn't be an issue tho, since the OpenVINO Model Optimizer would handle the data type (you need to specify from the conversion command).
I believe its better to keep the consistency of the precision instead of mixing them in your native model. In OpenVINO you can only use one data type per inference.
For example, in the case of VPU, it strictly requires FP16.

Your custom trained model might have something called overfitting (the model fits too much noise from the data) or underfitting(the model does not capture the underlying trend of data and does not fit the data well enough). If these are the reasons for the strange result, it depends on your Model Evaluation (Classification Error Matrix/Regression Error Matrix, etc) which **not related with OpenVINO**, instead you'll need to review/research your native model.

By default, OpenVINO should support the operations using native onnx model as long as the topology is supported and the model suffice what OpenVINO requires to infer. So, I checked your native model and finds that your model is actually using dynamic shape which currently not supported by OpenVINO.

**This conclude** the reason why your model is actually having a strange result.
I suggest for you to check out the validated and supported onnx topology [here](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_ONNX.html#supported_public_onnx_topologies).




![native_benchmark](https://user-images.githubusercontent.com/79643476/136438731-25d0eafa-1889-4c11-a1a8-009341f9c6f1.png)

**Comment by Iffa-Meah** [comment]
Closing issue, feel free to re-open or start a new issue if additional assistance is needed.
